### PR TITLE
Deploy RDMA CNI as a part of SR-IOV Operator

### PR DESCRIPTION
**Pull request by e0ne**

### CODE REVIEW

File: hack/env.sh
Comment (ykulazhenkov) [inline hack/env.sh:4]: Do you want to provide a way to skip deployment of the cni? I thin this construction will always use default vaule even if env var is explicitly set to an empty string. In my PR I used a different contruction https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/4b3a309a38d419a86dc7edb0a1bbb78fb13f71ac/hack/env.sh#L5. The difference is `:-` vs `-`. With construction from my PR it is possible to explicitly set the value to an empty string.

File: hack/env.sh
Comment (ykulazhenkov) [inline hack/env.sh:13]: Same question here. We can change the check like this https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/4b3a309a38d419a86dc7edb0a1bbb78fb13f71ac/hack/env.sh#L15. To allow empty string as a value.

File: hack/env.sh
Comment (e0ne) [reply hack/env.sh:4]: It makes sense. I'll update my PR

Comment (e0ne) [reply hack/env.sh:4]: done

File: hack/env.sh
Comment (e0ne) [reply hack/env.sh:13]: done

**Review by ykulazhenkov — APPROVED** [review]
LGTM

File: bindata/manifests/daemon/daemonset.yaml
Comment (zeeke) [inline bindata/manifests/daemon/daemonset.yaml:87]: Can we make this block optional? something like:
```suggestion
       {{- if ne .RDMACNIImage "" }}
        - name: rdma-cni
          image: {{.RDMACNIImage}}
          command:
            - /bin/sh
            - -c
            - cp /usr/bin/rdma /host/opt/cni/bin/
          securityContext:
            privileged: true
          resources:
            requests:
              cpu: 10m
              memory: 10Mi
          volumeMounts:
            - name: cnibin
              mountPath: /host/opt/cni/bin\
          {{- end }}
```

This would significantly reduce (or at least postpone) the effort to maintain this in openshift.

Comment (e0ne) [reply bindata/manifests/daemon/daemonset.yaml:87]: done

**Review by zeeke — APPROVED** [review]
LGTM

File: bindata/manifests/daemon/daemonset.yaml
Comment (SchSeba) [inline bindata/manifests/daemon/daemonset.yaml:95]: nit: please can the if statement be the same as `{{- if .OVSCNIImage }}`

File: bindata/manifests/daemon/daemonset.yaml
Comment (SchSeba) [inline bindata/manifests/daemon/daemonset.yaml:99]: please let's not have a copy like this here.

can we create an entrypoint.sh and use it in the image for the rdma-cni repo (I can review and approve it)

the same way we do for the sriov-cni

File: hack/env.sh
Comment (SchSeba) [inline hack/env.sh:19]: please do something like we did for the ovs-cni

```
# ensure that OVS_CNI_IMAGE is set, empty string is a valid value
OVS_CNI_IMAGE=${OVS_CNI_IMAGE:-}
```

**Review by SchSeba — COMMENTED** [review]
sorry for the super late review.

I add some small comments

File: bindata/manifests/daemon/daemonset.yaml
Comment (e0ne) [reply bindata/manifests/daemon/daemonset.yaml:95]: done

File: hack/env.sh
Comment (e0ne) [reply hack/env.sh:19]: done

File: hack/env.sh
Comment (SchSeba) [inline hack/env.sh:17]: $RDMA_CNI_IMAGE= -> RDMA_CNI_IMAGE=

File: bindata/manifests/daemon/daemonset.yaml
Comment (SchSeba) [inline bindata/manifests/daemon/daemonset.yaml:105]: please also add
```
- name: os-release
              mountPath: /host/etc/os-release
              readOnly: true
```

**Review by SchSeba — COMMENTED** [review]
just two small comments and we should be good to go nice work

File: hack/env.sh
Comment (e0ne) [reply hack/env.sh:17]: done

File: bindata/manifests/daemon/daemonset.yaml
Comment (e0ne) [reply bindata/manifests/daemon/daemonset.yaml:105]: done

**Review by SchSeba — APPROVED** [review]
LGTM

/hold

please lets wait for https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/700 and release cut before we merge this

**Comment by adrianchiris** [comment]
@e0ne please rebase so we can give it a final review and merge.

**Comment by e0ne** [comment]
> @e0ne please rebase so we can give it a final review and merge.

done

File: hack/env.sh
Comment (rollandf) [inline hack/env.sh:7]: Please align indentation.

File: hack/env.sh
Comment (rollandf) [inline hack/env.sh:19]: Please align indentation.

**Review by adrianchiris — COMMENTED** [review]
LGTM, please align indent as @rollandf commented and we can merge.

**Review by adrianchiris — APPROVED** [review]

File: deployment/sriov-network-operator-chart/values.yaml
Comment (adrianchiris) [inline deployment/sriov-network-operator-chart/values.yaml:106]: please update helm chart update script : https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/2fed72e8ce4aa89dbcd112d03aa2ff7139903591/hack/release/chart-update.sh#L47

**Review by adrianchiris — COMMENTED** [review]
@e0ne can you rebase against master ?

File: deployment/sriov-network-operator-chart/values.yaml
Comment (e0ne) [reply deployment/sriov-network-operator-chart/values.yaml:106]: @adrianchiris  done

**Review by SchSeba — APPROVED** [review]
LGTM

nice work!

**Comment by SchSeba** [comment]
/hold 

Hi @e0ne please check the k8s CI looks like the helm deploy doesn't work

**Comment by SchSeba** [comment]
please add the image here

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/hack/run-e2e-conformance-virtual-cluster.sh#L387

**Comment by e0ne** [comment]
> please add the image here
> 
> https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/hack/run-e2e-conformance-virtual-cluster.sh#L387

why do we need to to it?

**Comment by SchSeba** [comment]
Hi @adrianchiris can you please approve so we can merge this one?

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**
